### PR TITLE
LE Audio: Broadcast audio list instead of array

### LINKED ISSUE #43767: LE Audio: Broadcast sink/source use list of streams instead of array
**Is your enhancement proposal related to a problem? Please describe.**
The broadcast sink/source API requires a long-living array of pointers to streams to work, but can be improved by using a list. 

**Describe the solution you'd like**
Instead of using an array of pointers of streams, the sink/source can use the existing `node` in the `bt_audio_stream` to add them to a list instead, as is done for the unicast group. 

This will also remove the requirement that the array of pointers of stream shall remain static. 

**Describe alternatives you've considered**

**Additional context**
Add any other context or graphics (drag-and-drop an image) about the feature request here.

### PR DESCRIPTION
**Pull request by Thalley**
Modifies the broadcast source and sink APIs and implementations to use a slist instead of an array of pointers to hold the streams. 

fixes https://github.com/zephyrproject-rtos/zephyr/issues/43767

### CODE REVIEW

**Comment by Thalley** [comment]
Depends on https://github.com/zephyrproject-rtos/zephyr/pull/43709

**Comment by Thalley** [comment]
Rebased to fix merge conflicts

**Comment by Thalley** [comment]
Rebased to fix test failure

File: subsys/bluetooth/audio/broadcast_sink.c
Comment (MariuszSkamra) [inline subsys/bluetooth/audio/broadcast_sink.c:862]: Additionally, we should change the `bt_audio_stream` member `node` node to be prefixed with `_` -> `_node`. So that it would not be used by the user to put it on some other list :D

File: subsys/bluetooth/audio/endpoint.h
Comment (MariuszSkamra) [inline subsys/bluetooth/audio/endpoint.h:80]: Where's `sys_slist_init`?

File: subsys/bluetooth/audio/endpoint.h
Comment (MariuszSkamra) [inline subsys/bluetooth/audio/endpoint.h:99]: ditto

**Review by MariuszSkamra — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/audio/broadcast_sink.c
Comment (Thalley) [reply subsys/bluetooth/audio/broadcast_sink.c:862]: Agreed. I don't think we have any general guidelines for this in Zephyr (none that I am aware of), but the `_` does typically mean "internal". 

We do have a lot (a lot!) of fields in public structs that are internal and do not use the `_` prefix. 

We can easily fix that for ISO and LE Audio still, as those are experimental. Ideally we should be able to do this for all internal fields in public struct, as those should not be used by the applications, but I guess for the stable APIs we should still go through the proper process of changing those (if using `_` as a prefix is a convention we will start to follow

File: subsys/bluetooth/audio/endpoint.h
Comment (Thalley) [reply subsys/bluetooth/audio/endpoint.h:80]: `sys_slist_init` is defined as 
```c
static inline void sys_slist_init(sys_slist_t *list)
{
	list->head = NULL;
	list->tail = NULL;
}
```

and since the `bt_audio_broadcast_source`s and `bt_audio_broadcast_sink`s are statically defined, as well as `memset` on deletion, `sys_slist_init` provides no value. 

I can, however, still add it as part of the `create` function for semantic reasons if you like.

File: subsys/bluetooth/audio/endpoint.h
Comment (Thalley) [reply subsys/bluetooth/audio/endpoint.h:99]: See https://github.com/zephyrproject-rtos/zephyr/pull/44828#discussion_r880319450

File: subsys/bluetooth/audio/broadcast_sink.c
Comment (Thalley) [reply subsys/bluetooth/audio/broadcast_sink.c]: Added the "_" prefix for all nodes in LE Audio (not that many yet).

**Review by MariuszSkamra — DISMISSED** [review]

**Review by Casper-Bonde-Bose — DISMISSED** [review]
LGTM

**Comment by carlescufi** [comment]
@Thalley please rebase

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Comment by carlescufi** [comment]
@MariuszSkamra please take a look

**Review by asbjornsabo — COMMENTED** [review]
I am wondering about the first commit - it does not really do much, and it does not seem to fit neither the title nor the description it has.
Otherwise, this looks sane to me.

(Posting this as a comment, not a "request changes", as I may not be able to follow up and do not want to block the PR.)

**Review by MariuszSkamra — DISMISSED** [review]

**Comment by Thalley** [comment]
> I am wondering about the first commit - it does not really do much, and it does not seem to fit neither the title nor the description it has. Otherwise, this looks sane to me.
> 
> (Posting this as a comment, not a "request changes", as I may not be able to follow up and do not want to block the PR.)

Squashed the first commit. Not sure how it ended up like that :D

**Comment by Thalley** [comment]
Fixed build error after another commit was merged

**Review by asbjornsabo — DISMISSED** [review]

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Review by asbjornsabo — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**
